### PR TITLE
Rolemanagement

### PR DESCRIPTION
**Pull request by keerthi-mtalkz**

### CODE REVIEW

File: src/pages/role/update/[updateID].js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      <div className="w-full mb-4">
+        <label className="block">
+          <span className="text-default">Name</span>
+          <input
+            name="name"
+            type="text"
+            ref={register({ required: true })}
+            className="form-input mt-1 text-xs block w-full bg-white"
+            placeholder="Enter your name"
+            defaultValue={res.name}
```
Comment (gaurav-mtalkz) [inline src/pages/role/update/[updateID].js:168]: Why use defaultValue instead of value attribute?

File: src/pages/role/update/[updateID].js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+        width={48}
+        className="react-switch"
+      />
+        </label>
+      </div>
+ 
+      <div style={{ width: "300px" }}>
+      <Select
+        options={permissions}
+        placeholder="Select Permission ..."
```
Comment (gaurav-mtalkz) [inline src/pages/role/update/[updateID].js:212]: "Select Permissions"

File: src/pages/role/view/[viewID].js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                        Description
+                        </p>
+                       
+                    </div>
+                    <div className="inline-flex items-center text-base font-semibold text-gray-900 dark:text-white">
+                    <p className="text-sm font-medium text-gray-900 truncate dark:text-white">{res.description}</p>
+                    </div>
+                </div>
+            </li>
+        </ul>
```
Comment (gaurav-mtalkz) [inline src/pages/role/view/[viewID].js:105]: It is critical to show the list of permissions associated with the role

File: src/pages/role/addRole.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+      className="flex flex-col text-sm mb-4 lg:w-1/3"
+    >
+      {/*input*/}
+      <div className="w-full mb-4">
+        <label className="block">
+          <span className="text-default">Name</span>
+          <input
+            name="name"
+            type="text"
+            ref={register({ required: true })}
```
[the remark this answers is not in the document]
Comment (keerthi-mtalkz) [reply src/pages/role/addRole.js:114]: on top

[the remark this answers is not in the document]
Comment (gaurav-mtalkz) [reply src/pages/role/addRole.js:114]: So, we are using a different mechanism here as compared to other components?

**Review by gaurav-mtalkz — APPROVED** [review]

**Pull request merged**